test/Utilities/cachingoptimizer.jl

```diff
@@ -108,7 +108,11 @@ struct DummyConstraintAttribute <: MOI.AbstractConstraintAttribute end
 
 @testset "Mapping of variables" begin
     mock = MOIU.MockOptimizer(MOIU.UniversalFallback(MOIU.Model{Float64}()))
-    model = MOIU.CachingOptimizer(MOIU.UniversalFallback(MOIU.Model{Float64}()), mock)
+    model = MOIU.CachingOptimizer(
+        MOIU.UniversalFallback(MOIU.Model{Float64}()),
+        mock,
+    )
+    MOIU.attach_optimizer(model)
     x = MOI.add_variable(model)
     y = first(MOI.get(mock, MOI.ListOfVariableIndices()))
```

Comment:
@blegat it's breaking here without an explicit `attach_optimizer`. People doing stuff with the backend of JuMP before `optimize!` may run into errors.

Reply:
Which error ? In `AUTOMATIC` mode, we automatically attach instead of erroring.

Reply:
This example queries the inner optimizer of the cache directly.

Reply:
You shouldn't expect anything on the state of the inner optimizer in `AUTOMATIC` mode.
The only good reason users were accessing the inner optimizer was for optimizer attributes but that is not changed by this PR.